Detectors/ITSMFT/MFT/tracking/include/MFTTracking/TrackCA.h

```diff
@@ -117,58 +132,61 @@ inline void TrackCA::setPoint(const Float_t x, const Float_t y, const Float_t z,
     mLayer[mNPoints] = layer;
     mClusterId[mNPoints] = clusterId;
     mMCCompLabels[mNPoints] = label;
+    setClusterIndex(layer, clusterId, mNPoints);
   }
 }
 
-class TrackLTF final
+inline void TrackLTF::sort()
 {
- public:
-  TrackLTF() = default;
-  ~TrackLTF() = default;
-  void setPoint(const Float_t x, const Float_t y, const Float_t z, const Int_t layer, const Int_t clusterId, const MCCompLabel label, Bool_t& newPoint);
-  const Int_t getNPoints() const { return mNPoints; }
-  const std::array<Float_t, constants::mft::LayersNumber>& getXCoordinates() const { return mX; }
-  const std::array<Float_t, constants::mft::LayersNumber>& getYCoordinates() const { return mY; }
-  const std::array<Float_t, constants::mft::LayersNumber>& getZCoordinates() const { return mZ; }
-  const std::array<Int_t, constants::mft::LayersNumber>& getLayers() const { return mLayer; }
-  const std::array<Int_t, constants::mft::LayersNumber>& getClustersId() const { return mClusterId; }
-  const std::array<MCCompLabel, constants::mft::LayersNumber>& getMCCompLabels() const { return mMCCompLabels; }
+  // Orders elements along z position
 
- private:
-  Int_t mNPoints = 0;
-  std::array<Float_t, constants::mft::LayersNumber> mX = {-25., -25., -25., -25., -25., -25., -25., -25., -25., -25.};
-  std::array<Float_t, constants::mft::LayersNumber> mY = {-25., -25., -25., -25., -25., -25., -25., -25., -25., -25.};
-  std::array<Float_t, constants::mft::LayersNumber> mZ = {-120., -120., -120., -120., -120., -120., -120., -120., -120., -120.};
-  std::array<Int_t, constants::mft::LayersNumber> mLayer;
-  std::array<Int_t, constants::mft::LayersNumber> mClusterId;
-  std::array<MCCompLabel, constants::mft::LayersNumber> mMCCompLabels;
-  ClassDefNV(TrackLTF, 1);
-};
+  struct ClusterData {
+    Float_t x;
+    Float_t y;
+    Float_t z;
+    Float_t layer;
+    Float_t clusterId;
+    Float_t label;
```

Comment:
Why are these data members float (particularly, label cannot be float as it is assigned MCCompLabel)
Do you really need the label, given that you have id of original cluster (which in general may have multiple labels?)?

Reply:
Fixed.